lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py

```diff
@@ -520,206 +536,187 @@ def test_normal_catch_warnings_percentage_change(self, warning_list=None):
         self.assertTrue(any(warning_msg_iter in str(item)
                             for item in warning_list))
 
+
+class Test_crps_minimiser_wrapper_truncated_gaussian_distribution(
+        SetupInputs, EnsembleCalibrationAssertions):
+
+    """
+    Test minimising the CRPS for a normal distribution.
+    Either the ensemble mean or the individual ensemble realizations are used
+    as the predictors.
+    """
+    def setUp(self):
+        """Set up expected output."""
+        super().setUp()
+        self.expected_mean_coefficients = (
+            [0.000005, 1.543387, -0.514061, 0.939994])
+        self.expected_realizations_coefficients = (
+            [0.080978, 1.34056, -0.031015, 0.700256, -0.003556, 0.608326])
+
     """Test minimising the CRPS for a truncated_normal distribution."""
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate.",
-                          "Minimisation did not result in convergence"])
-    def test_basic_truncated_normal_mean_predictor(self):
+                          "The final iteration resulted in",
+                          "invalid value encountered in",
+                          "divide by zero encountered in"],
+        warning_types=[UserWarning, UserWarning, RuntimeWarning,
+                       RuntimeWarning])
+    def test_basic_mean_predictor(self):
         """
         Test that the plugin returns a numpy float value.
         The ensemble mean is the predictor.
         """
-        initial_guess = [5, 1, 0, 1]
-        initial_guess = np.array(initial_guess, dtype=np.float32)
-        cube = set_up_temperature_cube()
-
-        forecast_predictor = cube.collapsed("realization", iris.analysis.MEAN)
-        forecast_variance = cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-        truth = cube.collapsed("realization", iris.analysis.MAX)
-
         predictor_of_mean_flag = "mean"
+        distribution = "truncated gaussian"
 
         plugin = Plugin()
-        distribution = "truncated gaussian"
         result = plugin.crps_minimiser_wrapper(
-            initial_guess, forecast_predictor, truth, forecast_variance,
+            self.initial_guess_for_mean,
```

Comment:
A little hard to see in just the GitHub diff here, and in a sense not completely related to the changes @gavinevans is making - but if we're refactoring the unit tests to have a better structure, I think we could do much more of the setup for these tests in shared methods. (And in many of the other unit tests). In my opinion, keeping the setup separate from the rest of the assertions allows us to more clearly see what each of the unit tests are actually testing.

Reply:
Yes, I've updated the unit tests to factor out as much as possible into separate setup classes or methods, so as to limit the amount of code within each individual test.